flink-core/src/main/java/org/apache/flink/core/fs/FileSystem.java

```diff
@@ -55,6 +60,40 @@
 
 	private static final String HADOOP_WRAPPER_SCHEME = "hdwrapper";
 
+	private static final Logger LOG = LoggerFactory.getLogger(FileSystem.class);
+
+	/**
+	 * Create a SafetyNetCloseableRegistry for a Task. This method should be called at the beginning of the task's
+	 * main thread.
+	 */
+	public static void createFileSystemCloseableRegistryForTask() {
+		SafetyNetCloseableRegistry current = REGISTRIES.get();
+		if (null != current) {
+			if (!current.isClosed()) {
+				IOUtils.closeQuietly(current);
+				LOG.warn("Found open SafetyNetCloseableRegistry. Closed and replaced it.");
+			}
+		}
+		REGISTRIES.set(new SafetyNetCloseableRegistry());
+	}
+
+	/**
+	 * Create a SafetyNetCloseableRegistry for a Task. This method should be called at the end of the task's
+	 * main thread or when the task should be canceled.
+	 */
+	public static void disposeFileSystemCloseableRegistryForTask() {
+		SafetyNetCloseableRegistry ret = REGISTRIES.get();
+		if (null != ret) {
+			LOG.info("Closing SafetyNetCloseableRegistry.");
```

Comment:
Let's log something like "Ensuring all FileSystem streams are closed". Users will understand that better ;-)